### PR TITLE
docs: update architecture.md

### PR DESCRIPTION
**Pull request by bergwolf**
* match the current architecture of Kata Containers 2.0.
* fix static checker errors in docs
* drop docker installation guides

### CODE REVIEW

File: docs/design/architecture.md [hunk 1]
```diff
@@ -134,166 +119,32 @@ The only service running in the context of the initrd is the [Agent](#agent) as
...
-spawns the container process inside the VM, leveraging the `libcontainer` package.
-9. Start `kata-shim`, which will connect to the gRPC server socket provided by the `kata-proxy`. `kata-shim`  will spawn a few Go routines to parallelize blocking calls `ReadStdout()` , `ReadStderr()` and `WaitProcess()`. Both `ReadStdout()` and `ReadStderr()` are run through infinite loops since `kata-shim` wants the output of those until the container process terminates. `WaitProcess()` is a unique call which returns the exit code of the container process when it terminates inside the VM. Note that `kata-shim` is started inside the network namespace, to allow upper layers to determine which network namespace has been created and by checking the `kata-shim` process. It also creates a new PID namespace by entering into it. This ensures that all `kata-shim` processes belonging to the same container will get killed when the `kata-shim` representing the container process terminates.
-
-At this point the container process is running inside of the VM, and it is represented
-on the host system by the `kata-shim` process.
-
-![`kata-oci-create`](arch-images/kata-oci-create.svg)
```
Comment (liubin) [inline docs/design/architecture.md:206]: Hi, some media files(.png/.svg) is deleted from document, should delete the files itself ?

Comment (bergwolf) [reply docs/design/architecture.md:206]: yeah, it is removed

**Review by devimc — APPROVED** [review]
thanks @bergwolf

File: docs/design/architecture.md [hunk 2]
```diff
@@ -134,166 +119,32 @@ The only service running in the context of the initrd is the [Agent](#agent) as
...
-default this file is installed in the `/usr/share/defaults/kata-containers`
-directory and contains various settings such as the paths to the hypervisor,
+The runtime uses a TOML format configuration file called `configuration.toml`. By default this file is installed in the `/usr/share/defaults/kata-containers` directory and contains various settings such as the paths to the hypervisor,
 the guest kernel and the mini-OS image.
 
 Most users will not need to modify the configuration file.
 
-The file is well commented and provides a few "knobs" that can be used to modify
-the behavior of the runtime.
+The file is well commented and provides a few "knobs" that can be used to modify the behavior of the runtime.
```
Comment (jodh-intel) [inline docs/design/architecture.md:144]: I wonder if it's worth adding "and your chosen hypervisor" to the end of this sentence?

File: docs/design/architecture.md [hunk 3]
```diff
@@ -424,44 +254,15 @@ lifecycle management from container execution through the dedicated
...
-The Kubernetes\* execution unit is a pod that has specifications detailing constraints
-such as namespaces, groups, hardware resources, security contents, *etc* shared by all
-the containers within that pod.
-By default the Kubelet will send a container creation request to its CRI runtime for
-each pod and container creation. Without additional metadata from the CRI runtime,
-the Kata Containers runtime will thus create one virtual machine for each pod and for
-each containers within a pod. However the task of providing the Kubernetes pod semantics
-when creating one virtual machine for each container within the same pod is complex given
-the resources of these virtual machines (such as networking or PID) need to be shared.
+Kata Containers is an officially supported CRI-O and Containerd runtime. It is OCI compatible and therefore aligns with project's architecture and requirements.
```
Comment (jodh-intel) [inline docs/design/architecture.md:260]: > It is OCI compatible

It might be worth stating that this applies to 1.x only.

**Review by jodh-intel — APPROVED** [review]
Thanks for doing this @bergwolf !!

Just a few comments, so

lgtm

**Review by GabyCT — APPROVED** [review]

**Comment by jodh-intel** [comment]
Hi @bergwolf - could you update as the branch is now conflicted?

**Pull request merged**